src/parser/jobs/drk/ResourceSimulator.js

```diff
@@ -0,0 +1,599 @@
+			} mana for each wasted <ActionLink {...ACTIONS.DARK_ARTS}/>.  It's better to clip and use the mana than to cap.</Fragment>,
+			requirements: [
+				new Requirement({
+					name: 'Mana Utilization',
+					percent: ((Math.abs(this._totalSpentMana) / (this._totalGainedMana)) * 100),
+				}),
+			],
+		}))
+		this.checklist.add(new Rule({
+			name: 'Blood Utilization',
+			description: <Fragment>Capping on blood is a potency loss of about {BLOODSPILLER_BLOOD_POTENCY
+			} per {BLOODSPILLER_BLOOD_COST} blood.  Some cases of blood cap loss are unavoidable due to <ActionLink {...ACTIONS.THE_BLACKEST_NIGHT}/> usage.  Remember that <ActionLink {...ACTIONS.BLOODSPILLER}
+			/> does not reset your <ActionLink {...ACTIONS.SOULEATER}/>combo, consumes <ActionLink {...ACTIONS.DARK_ARTS}/>, and bypasses the major potency penalty from <ActionLink {...ACTIONS.GRIT}/>.</Fragment>,
+			requirements: [
+				new Requirement({
+					name: 'Blood Utilization',
+					percent: ((Math.abs(this._totalSpentBlood) / (this._totalGainedBlood)) * 100),
+				}),
+			],
+		}))
+		if (!this._darkArtsOpener) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.DARK_ARTS.icon,
+				content: <Fragment>
+					Missing a <ActionLink {...ACTIONS.DARK_ARTS}/> pre-pull for use during the opener is a large potency and enmity loss.  Using DA at around 6-7 seconds before pull gives ample time to use it as well as gain two pre-fight mana ticks.
+				</Fragment>,
+				severity: SEVERITY.MAJOR,
+				why: <Fragment>
+					Mana was logged at 100% when the fight started, which implies that a <ActionLink {...ACTIONS.DARK_ARTS}/> was not used pre-pull.  Ignore this if mana was at 100% for some other reason.
+				</Fragment>,
+			}))
+		}
+		if (this._droppedTBNs > 0) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.THE_BLACKEST_NIGHT.icon,
+				content: <Fragment>
+					One or more <ActionLink {...ACTIONS.THE_BLACKEST_NIGHT}/> applications did not pop, and thus did not generate blood.  This equates to throwing away the mana you could have spent on a <ActionLink {...ACTIONS.DARK_ARTS}/>.
+				</Fragment>,
+				severity: SEVERITY.MAJOR,
+				why: <Fragment>
+					You missed out on {this._droppedTBNs * 50} blood ({BLOODSPILLER_BLOOD_POTENCY} potency) or {this._droppedTBNs * 140} potency of Dark Arts buffs.
+				</Fragment>,
+			}))
+		}
+		if (this._noDACarve > 0) {
+			this.suggestions.add(new Suggestion({
+				icon: ACTIONS.CARVE_AND_SPIT.icon,
+				content: <Fragment>
+					One or more <ActionLink {...ACTIONS.CARVE_AND_SPIT}/> was used without <ActionLink {...ACTIONS.DARK_ARTS}/>
+					.  Even though this generates the same amount of mana as <ActionLink {...ACTIONS.SYPHON_STRIKE}/>, it loses out on 300 potency, a massive part of your damage as a DRK.
+				</Fragment>,
+				severity: SEVERITY.MAJOR,
+				why: <Fragment>
+					You missed out on {this._noDACarve * 350} potency due to {this._noDACarve} due to no buff carves.
```

Comment:
a) Magic numbers, b) this multiplies by 350 while the content text says 300 - which is correct?

Reply:
100 gets upgraded to 450; 350 difference. getting cleaned up